src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java

```diff
@@ -108,9 +108,10 @@ public void handle(Endpoint<?> endpoint) {
         HttpTracingUtils.setCommonTags(span, routingContext);
 
         try {
-            if (messageConverter != null) {
-                records = messageConverter.toKafkaRecords(topic, partition, routingContext.getBody());
+            if (messageConverter == null) {
+                return;
```

Comment:
Quick look, I don't think it's going to work. With the empty list we are going to procede anyway and having the possibility to reach line 170 where we are sending back a response to the client. Of course, it doesn't mean that empty records collection works. You should create a test with null messageConverter if possible and then finding a good way to send a response to the HTTP client that could make more sense. Anyway, it should be done in a different PR because I don't see it related to the spotbugs version, or?

Reply:
@stanlyDoge there is still this comment. Is removing this code that causes spotbugs to warn?

Reply:
I think it is stale. If you take a look at the updated code it is
```
if (messageConverter == null) {
   // this never happens, but let SB be happy
   sendCode500Response();
} else {
   //original code
   ...
}
```